plugin/js/showpass-custom.js

```diff
@@ -198,237 +256,337 @@
 			if (referrer) {
-				let url = new URL(iFrame.src);
 				url.searchParams.append("parent_document_referrer", referrer);
-				iFrame.src = url.toString();
 			}
-		}
 
-		/*
-		* Decorate iFrame for GA cross domain tracking
-	    * This only works for pop ups, for embedded calendar we need to use a watcher
-		*/
-		const mutationObserver = new MutationObserver(function(mutations) {
-			mutations.forEach(function(mutation) {
-				if (mutation.target.className.includes('showpass-widget-body') && document) {
+			// 3. Add current page query parameters
+			// This is REQUIRED for the checkout widget to work properly with Affirm redirects
+			const currentUrl = new URL(window.location.href);
+			const queryParams = currentUrl.searchParams;
 
-					let iFrame = document.getElementById('showpass-widget');
-
-					// if query params already exist, exit
-					let queryParams = new URLSearchParams(iFrame.src);
-					if (
-						queryParams.get("parent_client_id") ||
-						queryParams.get("parent_session_id") ||
-						queryParams.get("parent_document_referrer")
-					) {
-						return;
+			if (queryParams.toString()) {
+				queryParams.forEach((value, key) => {
+					// Check if parameter already exists in iframe src to avoid duplicates
+					if (!url.searchParams.has(key)) {
+						url.searchParams.append(key, value);
 					}
-					decorateIframe(iFrame);
+				});
+			}
+
+			// Update iframe src and mark as decorated
+			iFrame.src = url.toString();
+			iFrame.dataset.decorated = "true";
+
+			console.log(
+				"Decorated iframe with tracking parameters:",
+				url.toString()
+			);
+		};
+
+		/**
+		 * Sets up observer to watch for ANY Showpass iframe anywhere in the document
+		 * Handles both popup widgets and embedded widgets
+		 */
+		const setupShowpassIframeObserver = () => {
+			
+			// initial scan for any existing Showpass iframes
+			const existingIframes = document.querySelectorAll('iframe[src*="showpass.com"]');
+			existingIframes.forEach(iframe => {
+				if (!iframe.dataset.decorated) {
+					setTimeout(() => decorateIframe(iframe), 100);
 				}
 			});
-		});
 
-		mutationObserver.observe(document.documentElement, { attributes: true });
+			let processingMutations = false;
+			let pendingMutations = [];
+			
+			const documentObserver = new MutationObserver((mutations) => {
+				pendingMutations.push(...mutations);
 
-		let calendarDIV = document.getElementById("showpass-calendar-widget");
-		if (calendarDIV) {
-			/** Wrap IFrame for embeded calendar */
-			function wrapIFrame(iFrame) {
-				clearInterval(findIFrameInterval);
-				decorateIframe(iFrame);
-			}
-			const findIFrameInterval = setInterval(findIFrame, 100);
-			function findIFrame() {
-				let iFrame = document.getElementsByClassName("showpass-widget-iframe");
-				if (iFrame && iFrame[0] && iFrame[0].src) {
-				wrapIFrame(iFrame[0]);
-			}
-		}
+				if (processingMutations) {
+					return;
+				}
+				
+				processingMutations = true;
+
+				setTimeout(() => {
```

Comment:
why are we delaying this executation by 50ms?

Reply:
I initially added it due to going through the list of performance suggestions here: https://stackoverflow.com/questions/31659567/performance-of-mutationobserver-to-detect-nodes-in-entire-dom
But I'm now realizing that I read this wrong and the solution doesn't make sense as it's not referring to `processing is complex [due to the number of mutations]`, it's just referring to the processing callback itself which in our case is pretty simple, and the delays would just make it seem slower. Thanks for catching this one.

After going through the list again I missed the querySelectorAll suggestion so will add that one in.